src/lab10.py

```diff
@@ -166,7 +166,7 @@ def XMLHttpRequest_send(self, method, url, body):
         full_url = resolve_url(url, self.tab.url)
         if not self.tab.allowed_request(full_url):
             raise Exception("Cross-origin XHR blocked by CSP")
-        headers, out = request(full_url, self.tab.url, payload=body)
```

Comment:
Do named parameters not work in this case?

Reply:
Well, ugh, the compiler compiles named arguments to passing a dictionary as the last argument, but when it compiles the function being called it doesn’t know which arguments will be passed by keyword. In principle I can compile our functions and library functions differently, so we can keep using keyword arguments, but I’ve been thinking we need to refactor the networking part of the book anyway so I’d rather punt the keyword argument question for now.